### PR TITLE
[TE] Mock event generator for demo purposes

### PR DESCRIPTION
**Pull request by apucher**
Pseudo-random distribution-based generator for events similar to mock data source
- configuration via detector.yml, defaults to on for demo
- one-time execution at startup in backend with db-persistence and deduplication
- random event name generator (for kishore)

Screen Shot 2020-05-11 at 9 27 35 PM

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/MockEventsLoaderConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.List;
+
+public class MockEventsLoaderConfiguration {
+    public static class EventGeneratorConfig {
+        String type;
+        String arrivalType = MockEventsLoader.DIST_TYPE_EXPONENTIAL;
+        double arrivalMean;
+        String durationType = MockEventsLoader.DIST_TYPE_FIXED;
+        double durationMean = 86400000;
+        int seed = 0;
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/MockEventsLoaderConfiguration.java:16]: Can you include a one-line comment for these fields?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
+package org.apache.pinot.thirdeye.anomaly.events;
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java:1]: nit: license

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import org.apache.pinot.thirdeye.anomaly.MockEventsLoaderConfiguration;
+import org.apache.pinot.thirdeye.datalayer.bao.EventManager;
+import org.apache.pinot.thirdeye.datalayer.dto.EventDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+
+public class MockEventsLoader implements Runnable {
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java:14]: consider including a javadoc comment about this class

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    Set<EventDTO> dedup(Collection<EventDTO> generated, Collection<EventDTO> existing) {
+        Set<EventDTO> sorted = new TreeSet<>(MOCK_EVENT_COMPARATOR);
+        sorted.addAll(generated);
+        existing.forEach(sorted::remove);
+        return sorted;
+    }
+
+    AbstractRealDistribution makeDist(String type, double param, int seed) {
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/events/MockEventsLoader.java:80]: This is really cool! Can you include a comment about the technique in the code?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/MockEventsLoaderConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
+package org.apache.pinot.thirdeye.anomaly;
```
[the remark this answers is not in the document]
Comment (apucher) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/MockEventsLoaderConfiguration.java:20]: fixed

**Pull request merged**